File: 3451-string-compression-iii/3451-string-compression-iii.py
class Solution:
    def compressedString(self, word: str) -> str:
        result=""
        i=0
        s=set(list(word))
        print(s)
        count=0
        # for j in s:
        while i<len(word)-1:
            if word[i]==word[i+1]:
                count+=1
                i+=1
                if count==9:
                    result=result+"9"+word[i]
                    count=0
            else:
                result=result+str(count+1)+word[i]
                i+=1
                count=0
                
        result+=str(count+1)+word[i]
        return result

File: 3451-string-compression-iii/3451-string-compression-iii.java
public class Solution {
    public String compressedString(String word) {
        StringBuilder result = new StringBuilder();
        int i = 0;
        int count = 0;

        // Loop through the string
        while (i < word.length() - 1) {
            if (word.charAt(i) == word.charAt(i + 1)) {
                count++;
                i++;
                if (count == 9) {
                    result.append(9).append(word.charAt(i));
                    count = 0;
                }
            } else {
                result.append(count + 1).append(word.charAt(i));
                i++;
                count = 0;
            }
        }
        
        // Append the last character and its count
        result.append(count + 1).append(word.charAt(i));

        return result.toString();
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        String compressed = solution.compressedString("aaabbbccaa");
        System.out.println(compressed);  // Output: "3a3b2c2a"
    }
}
